libdebug/architectures/amd64/amd64_syscall_hijacker.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is part of libdebug Python library (https://github.com/libdebug/libdebug).
+# Copyright (c) 2024 Gabriele Digregorio. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for details.
+#
+
+from typing import Any, Callable
+from libdebug.state.thread_context import ThreadContext
+from libdebug.architectures.syscall_hijacking_manager import SyscallHijackingManager
+
+
+class Amd64SyscallHijacker(SyscallHijackingManager):
+    """Class that provides syscall hijacking for the x86_64 architecture."""
+
+    # Allowed arguments for the hijacker
+    allowed_args : set[str] = {
```

Comment:
Or we should add the corresponding register names here: `rax, rdi, rsi...` too, since the API `syscall_argx` was originally intended just for internal use.

Reply:
The syscall_argx function was presented as the 'official' method for interacting with the arguments within a syscall hook callback. This is documented in the README, in tests, and in the official talk slides, among other resources. 
Is it redundant? Oh, yep. 
However, if we keep this option in the callback, we must accept that these arguments are necessary.
We can change this choice everywhere but need to make a final decision. The API has been changed too many times already.